component/src/main/ballerina/ballerina/net/reliable/httpr/httpr.bal

```diff
@@ -0,0 +1,373 @@
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, "put") ;
+        return storeMessage(storableStream, generateMBConfiguration());
+    }
+    action patch (string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, "patch") ;
+        return storeMessage(storableStream, generateMBConfiguration());
+    }
+    action delete (string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, "delete") ;
+        return storeMessage(storableStream, generateMBConfiguration());
+    }
+    action options (string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, "options") ;
+        return storeMessage(storableStream, generateMBConfiguration());
+    }
+    action forward (string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, "forward") ;
+        return storeMessage(storableStream, generateMBConfiguration());
+    }
+    action execute (string HTTPVerb, string path, http:Request req) (http:Response, http:HttpConnectorError) {
+        blob storableStream = convertToBlob(req, connectorOptions, serviceUri, path, HTTPVerb) ;
+        return storeMessage(storableStream, generateMBConfiguration());
+    }
+}
+
+function storeMessage(blob storableStream, processor:GuaranteedProcessor guaranteedProcessor)(http:Response, http:HttpConnectorError) {
+
+    http:Response response = {};
+    string destinationName = guaranteedProcessor.destinationName;
+
+    // store the message
+    try {
+        transaction with retries(0) {
+            guaranteedProcessor.store(guaranteedProcessor.config, destinationName, storableStream);
+            log:printDebug("message successfully stored in persistent store");
+            response.setJsonPayload({status:"Message successfully received into Ballerina"});
+            response.setStatusCode(202);
+        }
+    } catch (error r) {
+        log:printError("error when storing the message. " + r.msg);
+        response.setJsonPayload({status:"Message failed to receive into Ballerina"});
+        response.setStatusCode(503);
+    }
+    return response, null;
+}
+
+function convertToBlob(http:Request request, http:Options options, string serviceUri, string path, string httpMethod) (blob) {
+    string encodedPayload = util:base64Encode(request.getStringPayload());
+    HttpEPInvoke epInvoke = {request:request, serviceUrl:serviceUri, path:path, httpMethod:httpMethod, connectorOptions:options, payload:encodedPayload};
+    var epInvokeJson,_ = <json> epInvoke;
+    string epInvokeString = epInvokeJson.toString();
+    blob serialized = epInvokeString.toBlob("UTF-8");
```

Comment:
move "UTF-8" to a constant, and reuse it everywhere.

Reply:
Added